### ISSUE TITLE
Allow bridged accounts to subscribe to Bluesky blocklists 

### ISSUE THREAD

**Issue by jonpincus** [snarfed/bridgy-fed#1632]
Subscribing to blocklists is key for reducing unwanted interaction on Bluesky -- including from harassers.  The increase in harassment over the last week makes this functionality even more critical.

Not sure just how this would work from a user interface perspective, but I want @thenexusofprivacy.infosec.exchange.ap.brid.gy to be able to subscribe to a moderation list like https://bsky.app/profile/blacksky.app/lists/3kaaqcnnxi72w   Bluesky offers two subscription options -- muting and blocking -- so a low-tech (but not particularly usable) option is to send a DM to the Bridgy Fed account 

subscribe  https://bsky.app/profile/blacksky.app/lists/3kaaqcnnxi72w [block | mute]
unsubscribe https://bsky.app/profile/blacksky.app/lists/3kaaqcnnxi72w

**Comment by snarfed** [comment]
Great feature request. Thank you for filing!

That kind of DM command definitely seems like the most likely path, at least for now. Not ideal, but UX for doing things across the bridge like this will always be a bit awkward at best. At least it's an ongoing subscription that automatically gets updates to the list, unlike most fediverse software that only has one-time imports at best.

**Comment by snarfed** [comment]
This is similar to #1406 for blocking individual users across the bridge.

**Comment by jonpincus** [comment]
> Mutes have to be handled privately/largely internally by Bridgy Fed.
> I'm not sure what would go into that, but I think it would take a significant effort and possibly runtime cost.

I think it's fine if mutes aren't handled initially (or any time soon), blocking's the key here.

**Comment by Tamschi** [comment]
I agree, yes. It's still very useful even if the list block implementation is so incomplete that a few users' programmatic interactions can still be bridged.

(If we're talking about similar block lists, there currently are a few users trying to retaliate by putting every subscriber to those lists on their own often mislabeled or defamatory lists. It's not bothering me much since the less overlap my and their circles have the better, though.)

**Comment by snarfed** [comment]
Mechanically, this is done in Bluesky by creating a `bsky.app.graph.listblock` record in the blocker's repo. https://docs.bsky.app/docs/tutorials/user-lists#blockunblock-a-list

cc @anujahooja

**Comment by Tamschi** [comment]
@snarfed Also note that user and moderation lists seem to be (semantically, at least) distinct, so the https://bsky.app/profile/snarfed.org/lists/3ku4roau2e42l you mentioned in #2178 should probably not be blockable this way.

**Edit:**

The distinction is:

`purpose: "app.bsky.graph.defs#modlist"` for a list that's used for blocking vs.
`purpose: "app.bsky.graph.defs#curatelist"` for lists that can be pinned like/as feeds.

I think historically there may also be lists that don't have this property, and _iinm_ they should act as modlists since that list purpose is older.

**Comment by snarfed** [comment]
@Tamschi right! There's also `#referencelist`: _A list of actors used for only for reference purposes such as within a starter pack._

I'm familiar with the list purpose field, but I interpreted it more as informational than a strict constraint. Is there something specific you've seen that says curate and reference lists can't/shouldn't be blocked?

(I know the bsky.app UI shows a _Subscribe_ button on mod lists and not the others.)

**Comment by snarfed** [comment]
Interesting. Blocking a `#modlist` or `#curatelist` doesn't have any effect on bsky.app. Haven't tested other clients yet.

**Comment by snarfed** [comment]
Done, and deployed. Feel free to try! Docs: https://fed.brid.gy/docs#blocklist

**Comment by Tamschi** [comment]
> […] Is there something specific you've seen that says curate and reference lists can't/shouldn't be blocked?

Not outside of my concerns about potential user confusion, but I think there could be a problem in case users can't undo these through normal means after migrating off the bridge (short of using PDSls to find and delete the records). Of course I'd hope that these functions are already implemented in a way that works fine in light of these edge cases, though.
